### PR TITLE
Fix Okta Authentication and Validation - Ingestion

### LINKED ISSUE #2913: Ingestion - Token expires without renewing
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
Ingestion
**Describe the bug**
One of the users faced an `Expired Token` error
The Authentication from the Ingestion side does not handle the renewal of tokens.

**Expected behavior**
The Auth part on ingestion should renew the token if it nears expiry

### LINKED ISSUE #2953: Okta Authentication - Ingestion
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
Ingestion
**Describe the bug**
Okta Authentication from ingestion doesn't handle the creation and validation of access token

### PR DESCRIPTION
**Pull request by ayush-shah**
### Describe your changes :

fixes #2953 
fixes #2913
#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@harshach @pmbrull @vivekratnavel

### CODE REVIEW

File: ingestion/examples/auth_examples/okta_example.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "config": {}
+  },
+  "metadata_server": {
+    "type": "metadata-server",
+    "config": {
+      "api_endpoint": "http://localhost:8585/api",
+      "auth_provider_type": "okta",
+      "client_id": "{client_id}",
+      "org_url": "{Issuer URI}",
+      "private_key": "Public and Private Keypair",
```
Comment (harshach) [inline ingestion/examples/auth_examples/okta_example.json:19]: we should provide an example what is this private_key looks like

File: ingestion/src/metadata/ingestion/api/workflow.py [hunk 2]
```diff
@@ -162,15 +162,16 @@ def execute(self):
             if hasattr(self, "sink"):
                 self.sink.write_record(processed_record)
                 self.report["sink"] = self.sink.get_status().as_obj()
+            if hasattr(self, "bulk_sink"):
```
Comment (harshach) [inline ingestion/src/metadata/ingestion/api/workflow.py:165]: this should be outside of for loop

Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/api/workflow.py]: Resolved this.

File: ingestion/src/metadata/ingestion/ometa/ometa_api.py [hunk 3]
```diff
@@ -155,7 +156,7 @@ def __init__(self, config: MetadataServerConfig, raw_data: bool = False):
             base_url=self.config.api_endpoint,
             api_version=self.config.api_version,
             auth_header="Authorization",
-            auth_token=self._auth_provider.auth_token(),
+            auth_token=asyncio.run(self._auth_provider.auth_token()),
```
Comment (harshach) [inline ingestion/src/metadata/ingestion/ometa/ometa_api.py:159]: do we need aysncio here?

**Review by harshach — APPROVED** [review]

**Pull request merged**